src/main/java/dk/kb/discover/webservice/OpenApiResource.java

```diff
@@ -0,0 +1,117 @@
+package dk.kb.discover.webservice;
+
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import dk.kb.util.Resolver;
+import dk.kb.util.yaml.YAML;
+import org.yaml.snakeyaml.Yaml;
+
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class OpenApiResource {
+
+    public static final String APPLICATION_YAML = "application/yaml";
+
+    /**
+     * Pattern to allow search-replace for variabels defined as ${config.yaml.path} in OpenAPI specifications.
+     * Everything after 'config.' is treated as a path to an entry in the backing configuration.
+     */
+    private static final Pattern CONFIG_REPLACEMENT= Pattern.compile("\\$\\{config\\.([^}]+)\\}");
+
+    /**
+     * Deliver the OpenAPI specification with substituted configuration values as a YAML file.
+     */
+    @GET
+    @Produces(APPLICATION_YAML)
+    @Path("/ds-discover-openapi_v1.yaml")
+    public Response getYamlSpec() {
+        InputStream yamlStream = Resolver.openFileFromClasspath("ds-discover-openapi_v1.yaml");
+        String openApiSpec = YAML.parse(yamlStream).toString();
+
+        String replacedText = replaceConfigPlaceholders(openApiSpec);
+
+        Response.ResponseBuilder builder = Response.ok(replacedText).header("Content-Disposition", "inline; filename=ds-discover-openapi_v1.yaml");
+        return builder.build();
+    }
+
+    /**
+     * Deliver the OpenAPI specification with substituted configuration values as a JSON file.
+     */
+    @GET
+    @Produces(MediaType.APPLICATION_JSON)
+    @Path("/{path: (ds-discover-openapi_v1.json|openapi.json)}")
```

Comment:
Same as for the YAML-version: Can this be generified? Might be a bit tricky for `openapi.json`?

Reply:
I've generified it for the `{path}.json` part. However the `openapi.json` seems to be even more magical than discussed yesterday. OpenAPI still overwrites this, eventhough I've created this new endpoint. I'll look into which file it overwrites with. We might be lucky that it uses the one where replacements have already been done.

Reply:
Addendum: I've already created internal issue DRA-282 which is about  `/api-docs ` I believe that the magic for `/openapi.json` seems to fall in the same category. I'm thinking of adding the investigation into /openapi.json to that issue instead of diving into it now.

Reply:
I have investigated and `openapi.yaml` is generated (qualified guessing) from the annotated Java files. Yes, that means `ds-discover-openapi_v1.yaml` → Java → `openapi.yaml`. Maybe it can be overridden in `WEB-INF/web.xml`?

Where is the hardcoded `openapi.json` used?

Reply:
It is not used anywhere. It's only a OpenAPI thing to serve it there as well.

Reply:
I have added support for `openapi.yaml` to the internal ticket DRA-282.